symphony-bdk-core/src/main/java/com/symphony/bdk/core/client/loadbalancing/RegularLoadBalancedApiClient.java

```diff
@@ -0,0 +1,42 @@
+package com.symphony.bdk.core.client.loadbalancing;
+
+import com.symphony.bdk.core.client.ApiClientFactory;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiResponse;
+import com.symphony.bdk.http.api.Pair;
+import com.symphony.bdk.http.api.util.TypeReference;
+
+import org.apiguardian.api.API;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Concrete implementation of {@link LoadBalancedApiClient} where {@link #rotate()} is executed when
+ * {@link #invokeAPI(String, String, List, Object, Map, Map, Map, String, String, String[], TypeReference)} is called
+ * and if {@link #loadBalancingConfig} has the field stickiness set to false.
+ */
+@API(status = API.Status.INTERNAL)
+public class RegularLoadBalancedApiClient extends LoadBalancedApiClient {
+
+  public RegularLoadBalancedApiClient(BdkConfig config, ApiClientFactory apiClientFactory) {
+    super(config, apiClientFactory);
+  }
+
+  /**
+   * {@inheritDoc}
+   **/
+  @Override
+  public <T> ApiResponse<T> invokeAPI(String path, String method, List<Pair> queryParams, Object body,
+      Map<String, String> headerParams, Map<String, String> cookieParams, Map<String, Object> formParams, String accept,
+      String contentType, String[] authNames, TypeReference<T> returnType) throws ApiException {
+
+    if (!loadBalancingConfig.isStickiness()) {
+      rotate();
```

Comment:
If i use the external lb strategy does this mean for each API call we are going to perform 2 api calls (one to retrieve the fqdn first)?

Reply:
Yes indeed, except if we have `stickiness: true`. This is how it is implemented in the legacy BDK.